### PR TITLE
Improved search

### PR DESCRIPTION
**Pull request by BrunoBernardino**
This makes it possible to search for things like "Google - Padloc" with a term like "google padloc" instead of "google - padloc".

@MaKleSoft I'd love to push this on a 4.2.1, but I know we haven't really discussed it. Let me know your thoughts!

### CODE REVIEW

**Comment by BrunoBernardino** [comment]
The android build error seems to be caused by https://github.com/android-actions/setup-android/issues/378

**Comment by BrunoBernardino** [comment]
Now cypress is failing in CI due to https://github.com/cypress-io/cypress/issues/19299 (though locally all is fine). Will fix that too.

**Comment by BrunoBernardino** [comment]
Small proof I'm not crazy :sweat_smile: 

Screenshot from 2023-02-11 11-30-49

File: packages/app/src/elements/items-list.ts [hunk 1]
```diff
@@ -52,7 +52,19 @@ function filterByString(fs: string, rec: VaultItem) {
...
+        return content.search(escapeRegex(fs.toLowerCase())) !== -1;
+    }
+
+    for (const searchTerm of searchTerms) {
+        if (!content.includes(searchTerm)) {
+            return false;
+        }
+    }
+
+    return true;
```
Comment (MaKleSoft) [inline packages/app/src/elements/items-list.ts:67]: I think this should do the same thing in fewer lines (plus it's more readable):

```suggestion
    const searchTerms = escapeRegex(fs.toLowerCase()).split(" ").filter(Boolean);
    
    return searchTerms.every(searchTerm => content.includes(searchTerm));
```

**Review by MaKleSoft — CHANGES_REQUESTED** [review]
I like the idea! Just had a suggestion on how to make it more terse/readable. Other than that it should be good to merge.

**Comment by BrunoBernardino** [comment]
Now it's https://github.com/GitSquared/node-geolite2-redist/issues/26 ! I'll update `geolite2-redist`...

File: packages/app/src/elements/items-list.ts (hunk 1 above)
Comment (BrunoBernardino) [reply packages/app/src/elements/items-list.ts:67]: Ah, good one!

**Comment by BrunoBernardino** [comment]
I give up. The server isn't booting in CI for some weird reason and I can't figure that out. I'll just disable e2e tests in CI for now, but they're passing locally. FYI @MaKleSoft

File: cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { defineConfig } from "cypress";
```
Comment (BrunoBernardino) [inline cypress.config.ts:1]: Cypress 10 introduced a different config and file names. These changes are related to that.

**Review by MaKleSoft — APPROVED** [review]

**Pull request merged**